modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ExecutionServiceImpl.java

```diff
@@ -596,6 +643,212 @@ private QueryPlan prepareDml(SqlNode sqlNode, PlanningContext ctx) throws Valida
         return new MultiStepDmlPlan(template, queryFieldsMetadata(ctx, igniteRel.getRowType(), null));
     }
 
+    /** */
+    private QueryPlan prepareDdl(SqlNode sqlNode, PlanningContext ctx) {
+        assert sqlNode instanceof SqlDdl : sqlNode == null ? "null" : sqlNode.getClass().getName();
+
+        SqlDdl ddlNode = (SqlDdl)sqlNode;
+
+        if (ddlNode instanceof IgniteSqlCreateTable) {
+            IgniteSqlCreateTable createTblNode = (IgniteSqlCreateTable)ddlNode;
+
+            SqlIdentifier tblFullName = createTblNode.name();
+
+            CreateTableCommand createTblCmd = new CreateTableCommand();
+
+            createTblCmd.schemaName("PUBLIC"); // TODO: get from the context
```

Comment:
Seems not the only context could be a source for schemaName, but schemanName could also present in CREATE TABLE clause:
CREATE TABLE [schemaname.]tableName ...
As I understand for the second case we could extract it from` tblFullName.names`

Reply:
according to [this](https://ignite.apache.org/docs/latest/sql-reference/ddl#create-table) there is no such option like `schemaName`. A schema could be specified as part of the cache's template only, I suppose

Reply:
agree, my bad. In this case we should check that tablename is simple

Reply:
> should check that tablename is simple

currently it is handled by grammar. Are you OK if I just add an assertion here that would verify the grammar is not changed (names collection has exactly one element)

Reply:
fixed anyway

Reply:
regarding the ` [schemaName.]`... It appear that you were right. There is an ability in the old query engine to specify a schema name, so i've added this